client/api/src/client.rs

```diff
@@ -273,10 +273,14 @@ pub struct BlockImportNotification<Block: BlockT> {
 /// Summary of a finalized block.
 #[derive(Clone, Debug)]
 pub struct FinalityNotification<Block: BlockT> {
-	/// Imported block header hash.
+	/// Finalized block header hash.
 	pub hash: Block::Hash,
-	/// Imported block header.
+	/// Finalized block header.
 	pub header: Block::Header,
+	/// Previous explicitly finalized block header hash.
+	pub previous_finalized_hash: Block::Hash,
```

Comment:
Maybe we should instead have a `Arc<[Block::Hash]>` here? Where the first is the previous last finalized and the last is the currently finalized one.